tests/test_vec_frame_stack.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+from gym import spaces
+
+from stable_baselines3.common.vec_env.vec_frame_stack import VecFrameStack
+
+
+class BoxObsVecEnv:
+    """Custom Environment that produces box observations"""
+
+    metadata = {"render.modes": ["human"]}
+
+    def __init__(self, obs_space: spaces.Box):
+        self.num_envs = 4
+        self.action_space = spaces.Discrete(2)
+        self.observation_space = obs_space
+
+    def step_async(self, actions):
+        self.actions = actions
+
+    def step_wait(self):
+        obs = np.zeros((self.num_envs,) + self.observation_space.shape)
+        terminal_obs = np.zeros(self.observation_space.shape)
```

Comment:
the current tests don't fail if you remove your fix, no?

Reply:
Yep. Fixed now.